File: app/src/main/java/com/cashix/network/AuthInterceptor.java
package com.cashix.network;

import okhttp3.Interceptor;
import okhttp3.Request;
import okhttp3.Response;
import java.io.IOException;

public class AuthInterceptor implements Interceptor {
    private String authToken;

    public AuthInterceptor(String authToken) {
        this.authToken = authToken;
    }

    @Override
    public Response intercept(Chain chain) throws IOException {
        Request originalRequest = chain.request();

        // Add the Authorization header with the bearer token
        Request newRequest = originalRequest.newBuilder()
                .header("authorization", "Bearer " + authToken)
                .build();

        return chain.proceed(newRequest);
    }
}

File: app/src/main/java/com/cashix/kotlin/UI/shared/AuthInterceptor.kt
package com.cashix.kotlin.UI.shared

import android.util.Log
import okhttp3.Interceptor
import okhttp3.Request
import okhttp3.Response

class AuthInterceptor : Interceptor {
    private var authToken: String? = null

    fun updateToken(token: String) {
        authToken = token
    }

    fun getToken(): String? {
        return authToken
    }

    override fun intercept(chain: Interceptor.Chain): Response {
        val request: Request = chain.request().newBuilder()
            .addHeader("authorization", "Bearer $authToken")
            .build()
        return chain.proceed(request)
    }
}
